usecases/schema/class.go

```diff
@@ -151,61 +150,33 @@ func (h *Handler) UpdateClass(ctx context.Context, principal *models.Principal,
 	className string, updated *models.Class,
 ) error {
 	err := h.Authorizer.Authorize(principal, "update", "schema/objects")
-	if err != nil {
-		return err
-	}
-
-	initial := h.metaReader.ReadOnlyClass(className)
-	if initial == nil {
-		return ErrNotFound
-	}
-	mtEnabled, err := validateUpdatingMT(initial, updated)
-	if err != nil {
+	if err != nil || updated == nil {
 		return err
 	}
 
 	// make sure unset optionals on 'updated' don't lead to an error, as all
 	// optionals would have been set with defaults on the initial already
 	h.setClassDefaults(updated)
+	initial := h.metaReader.ReadOnlyClass(className)
 
-	if err := h.validateImmutableFields(initial, updated); err != nil {
-		return err
-	}
-	if err := h.parser.ParseClass(updated); err != nil {
-		return err
-	}
-
-	if err := h.validator.ValidateVectorIndexConfigUpdate(ctx,
-		initial.VectorIndexConfig.(schemaConfig.VectorIndexConfig),
-		updated.VectorIndexConfig.(schemaConfig.VectorIndexConfig)); err != nil {
-		return errors.Wrap(err, "vector index config")
-	}
-
-	if err := h.validator.ValidateInvertedIndexConfigUpdate(ctx,
-		initial.InvertedIndexConfig, updated.InvertedIndexConfig); err != nil {
-		return errors.Wrap(err, "inverted index config")
-	}
-	if err := validateShardingConfig(initial, updated, mtEnabled, h.clusterState); err != nil {
-		return fmt.Errorf("validate sharding config: %w", err)
-	}
+	// first layer of defense is basic validation if class already exits
+	if initial != nil {
+		_, err := validateUpdatingMT(initial, updated)
+		if err != nil {
+			return err
+		}
 
-	if err := replica.ValidateConfigUpdate(initial, updated, h.clusterState); err != nil {
-		return fmt.Errorf("replication config: %w", err)
-	}
+		if initial.ReplicationConfig.Factor != updated.ReplicationConfig.Factor {
+			return fmt.Errorf("updating replication factor is not supported yet")
+		}
```

Comment:
**question:**  is there is a reason not moving it to `validateImmutableFields` 🤔 ?

Reply:
Yes because these fields are basically not immutable, and they might change in the future